### PR TITLE
Enable `--experimental-local` for `wrangler pages dev`

### LINKED ISSUE #1947: `pages dev --experimental-local`
Once we've got https://github.com/cloudflare/wrangler2/pull/1940, it should be relatively easy (hopefully trivial!) to pipe through the `--experimental-local` option on `wrangler pages dev`.

### PR DESCRIPTION
**Pull request by GregBrimble**
Fixes #1947

### CODE REVIEW

File: packages/wrangler/src/miniflare-cli/assets.ts [hunk 1]
```diff
@@ -203,8 +210,8 @@ async function generateAssetsFetch(
 	};
 
 	return async (input: RequestInfo, init?: RequestInit) => {
-		const request = new MiniflareRequest(input, init);
-		return await generateResponse(request as unknown as Request);
+		// const request = new MiniflareRequest(input, init);
```
Comment (GregBrimble) [inline packages/wrangler/src/miniflare-cli/assets.ts:213]: This is going to break miniflare v2.

File: packages/wrangler/src/miniflare-cli/assets.ts [hunk 2]
```diff
@@ -4,10 +4,7 @@ import { createMetadataObject } from "@cloudflare/pages-shared/metadata-generato
 import { parseHeaders } from "@cloudflare/pages-shared/metadata-generator/parseHeaders";
 import { parseRedirects } from "@cloudflare/pages-shared/metadata-generator/parseRedirects";
 import { fetch as miniflareFetch } from "@miniflare/core";
-import {
-	Response as MiniflareResponse,
-	Request as MiniflareRequest,
-} from "@miniflare/core";
+import { Response as MiniflareResponse } from "@miniflare/core";
```
Comment (GregBrimble) [inline packages/wrangler/src/miniflare-cli/assets.ts:7]: Need to replace this (and probably a few others) with the Fetch API things newly exported from `@miniflare/tre`) (https://github.com/cloudflare/miniflare/pull/440).

File: fixtures/local-mode-tests/tests/headers.test.ts [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import path from "path";
 import { unstable_dev } from "wrangler";
 
-describe("worker", () => {
+describe.skip("worker", () => {
```
Comment (jrf0110) [inline fixtures/local-mode-tests/tests/headers.test.ts:4]: We still need to skip this suite?

Comment (GregBrimble) [reply fixtures/local-mode-tests/tests/headers.test.ts:4]: Yeah, it was being flakey in tests, so I'm just going to disable it for now.

Comment (jahands) [reply fixtures/local-mode-tests/tests/headers.test.ts:4]: Mind adding a comment above it saying this?

**Review by jrf0110 — APPROVED** [review]
LGTM!

**Pull request merged**